controllers/volumesnapshotbackup_controller.go

```diff
@@ -120,7 +138,24 @@ func (r *VolumeSnapshotBackupReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Check and add VSBs to queue until full
+	processed, err := r.setVSBQueue(&vsb, r.Log)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// no error but VSB queue is full
+	if !processed && err == nil {
+		r.Log.Info(fmt.Sprintf("requeuing vsb %v as max vsbs are being processed", vsb.Name))
+		return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
+	}
+
 	if !vsb.DeletionTimestamp.IsZero() {
+		// if batchingStatus is completed then processingVSBs has already been decremented
```

Comment:
Here the comment does not match with the condition, are we checking for `BatchingStatus == completed` or `BatchingStatus != completed` ?

Reply:
So `processingVSBs` already gets decremented once `SnapMoverBackupBatchingCompleted`. Therefore we would not want to decrement again if a VSB has that status but then starts to be deleted, because then it would be decremented twice for a single VSB